ddtrace/contrib/datadog_lambda/patch.py

```diff
@@ -0,0 +1,148 @@
+from importlib import import_module
+import os
+import signal
+
+from ddtrace import tracer
+from ddtrace.constants import ERROR_MSG
+from ddtrace.constants import ERROR_TYPE
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.internal.wrapping import unwrap
+from ddtrace.internal.wrapping import wrap
+
+
+log = get_logger(__name__)
+
+
+def crash_flush(_, __):
+    """
+    Tags the current root span with an Impending Timeout error and
+    finishes all remaining spans.
+
+    Shutdowns the tracer at the end.
+    """
+
+    root_span = tracer.current_root_span()
+    root_span.error = 1
+    root_span.set_tag_str(ERROR_MSG, "Datadog detected an Impending Timeout")
```

Comment:
```suggestion
    root_span.set_tag_str(ERROR_MSG, "Datadog detected an impending timeout")
```

nit (unless this is an official AWS or Datadog concept)

Reply:
This is an approved copy, it's a Datadog concept which Serverless is bringing up.